qmxgraph/_web_view.py

```diff
@@ -26,25 +30,18 @@ def __init__(self, *args, **kwargs):
         # It is common to use drag&drop with this widget
         self.setAcceptDrops(True)
 
-        self_ref = weakref.ref(self)
-
-        def post_load(ok):
-            self_ = self_ref()
-            if not self_ or not ok:
-                return
-
-            self_._loaded = True
-            self_._loading = False
+        self.loadFinished.connect(self._on_load_finished)
+        self.loadStarted.connect(self._on_load_started)
 
-        def during_load():
-            self_ = self_ref()
-            if not self_:
-                return
+    def _on_load_finished(self, ok):
+        if not ok:
+            return
 
-            self_.loading = True
```

Comment:
Oh my this was the cause of some headache... this should be `self_._loading = True` instead. Apparently it was never used by anyone, but the async nature of webengine made using this worthwhile in `load_and_wait`, but as this was never set correctly, `load_and_wait` was completely flaky. 😢